src/provisioning-ofono.c

```diff
@@ -336,8 +360,19 @@ static void add_mms_context_reply(DBusPendingCall *call, void *user_data)
 	DBusMessage *reply = dbus_pending_call_steal_reply(call);
 	DBusError err;
 	const char *path;
-	struct modem_data *modem;
+	struct modem_data *modem = NULL;
 	struct provisioning_data *prov_data;
+	GList *keys = g_hash_table_get_keys(modem_list);
+
+	while (keys != NULL) {
+		modem = g_hash_table_lookup(modem_list, keys->data);
+
+		if (modem->gprs_attached) {
+			break;
+		}
+		
+		keys = keys->next;
+	}
```

Comment:
What is the above code trying to do? Why is `gprs_attached` so important here, and what does it have to do with the modem for which `AddContext` call was made?